jsonschema-core.md

```diff
@@ -2071,71 +2071,44 @@ Omitting this keyword has the same assertion behavior as an empty schema.
-  further details.
-- *List*: Provides validation information in a flat list structure.
-
-Hierarchical: Provides validation information in a hierarchical structure that
-follows the evaluation paths generated while processing the schema.
-
-An implementation MUST provide the "flag" format and SHOULD provide at least one
-of the "list" or "hierarchical" formats. Implementations SHOULD specify in their
-documentation which formats they support.
-
-### Minimum Information
-
-Beyond the simplistic "flag" output, additional information is useful to aid in
-debugging a schema or instance. Each sub-result SHOULD contain the information
-contained within this section at a minimum.
-
-A single object that contains all of these components is considered an output
-unit.
-
-Implementations MAY elect to provide additional information.
-
-#### Evaluation path
-
-The evaluation path to the schema object that produced the output unit. The
-value MUST be expressed as a JSON Pointer, and it MUST include any by-reference
-applicators such as `$ref` or `$dynamicRef`.[^13]
-
-[^13]: The schema may not actually have a value at the location indicated by
-this pointer. It is provided as an indication of the traversal path only.
+The evaluation path is the set of keys, starting from the schema root, through
+which evaluation passes to reach the schema object that produced a specific
+result. The value MUST be expressed as a JSON Pointer, and it MUST include any
+by-reference applicators such as `$ref` or `$dynamicRef`.
 
 ```
 /properties/width/$ref/allOf/1
 ```
 
-Note that this pointer may not be resolvable by the normal JSON Pointer process
-due to the inclusion of these by-reference applicator keywords.
+Note that this pointer may not be resolvable on the root schema by the normal
+JSON Pointer process. It is intended as an indication of the traversal path
+only.
 
-The JSON key for this information is "evaluationPath".
+When represented in JSON, the key for this information MUST be "evaluationPath".
 
-#### Schema Location
+### Schema Location
 
-The absolute, dereferenced location of the schema object that produced the
-output unit. The value MUST be expressed using the canonical IRI of the relevant
-schema resource plus a JSON Pointer fragment that indicates the schema object
-that produced the output. It MUST NOT include by-reference applicators such as
-`$ref` or `$dynamicRef`.[^14]
+The schema location is the absolute, dereferenced location of the schema object
+that produced a result. The value MUST be expressed using the canonical IRI of
+the relevant schema resource plus a JSON Pointer fragment that indicates the
+schema object that produced the output. It MUST NOT include by-reference
+applicators such as `$ref` or `$dynamicRef`.[^14]
 
 [^14]: Note that "absolute" here is in the sense of "absolute filesystem path"
 (meaning the complete location) rather than the "absolute-IRI" terminology from
```

Comment:
I know this isn't a think that's actually changing in this PR, but I found this footnote confusing. What does filesystem have to do with this?

Reply:
This note was added as a clarification.

Ref: https://github.com/json-schema-org/json-schema-spec/commit/4a1035137f90ffff425bea257f48858dcd0d301c  (see Henry's commit note)

Reply:
I think if we changed the word from "absolute" to "non-relative" we can avoid to confusion and skip the footnote that IMO is even more confusing.

Reply:
What's the distinction that you see between those terms?

Reply:
They mean exactly the same thing, it just avoids the confusion caused by the common misunderstanding of what the term "absolute-IRI" means.

This value is an IRI (must contain a scheme and may have a fragment), so when we use the term "absolute", the reader may assume that that means it's an absolute-IRI (no fragment allowed), which is not what we meant. As best I can figure out, this footnote is trying to clarify that the use of "absolute" does not mean "absolute-IRI" and you should not assume that a fragment is not allowed.

I've taken to adding the term "non-relative" when talking about IRIs (must have a scheme and may have a fragment). When people say "absolute IRI", that's generally what they mean, but that's not technically correct and that leads to confusion. Using the term "non-relative" is explicit, clear, and doesn't have common misunderstandings that we need to clarify with a footnote.

Reply:
I think https://github.com/json-schema-org/json-schema-spec/pull/1429#discussion_r1323584899 may resolve this conversation as well.